salt/_modules/metalk8s.py

```diff
@@ -343,3 +344,14 @@ def format_slots(data):
             )
 
     return data
+
+
+def cmp_sorted(*args, **kwargs):
```

Comment:
I suppose at one point in time we will be able to remove this compat layer and fix the calling code.
Or are we gonna stay with those workaround forever?

Reply:
For me, we will need it for a long time since I do not see another way to do some sorting using a specific function in Jinja (Python3)

Reply:
So why can't we use `key`?
At one point in the future, we will be full python3 and there will be no need for `cmp` no?

Reply:
Because we can't easily in this context
The purpose for the downgrade is to sort node versions so you need to compare versions, if we want to use `key` instead of `cmp` we need to create dedicated objects for a version and store them... in the pillar (or have "something" (likely a function) to get a representation of a version that can be compared easily but in this case we cannot uise the `pkg.version_cmp` from salt that may have some logic we want here), for me, it's not possible easily in salt context